clang/lib/Sema/SPIRVBuiltins.td

```diff
@@ -62,6 +63,7 @@ class QualType<string _Name, bit _IsAbstract=0, bit _IsSigned=0> {
   // without access qualifiers.
   bit IsAbstract = _IsAbstract;
   bit IsSigned = _IsSigned;
+  bit IsFunctionCall = _IsFunctionCall;
```

Comment:
I'm not an expert in TableGen yet, but can we create a separate class to function calls generation, instead of making "Type" class represent also a "function"?

Reply:
That's a good suggestion, but the backend used to process the file is also handling OpenCLBuiltins.td. So if we change the structure for this, it will also impact this other part. It is doable and probably better on the long run, but so far I have been reluctant on touching this side until this part is stable.

If you look at OpenCLBuiltins.td file, you will see that there is common definition that we should ultimately merge together. The main problem is OpenCL operates under a strict language setting but it is much more vague for SYCL (because C++).

Reply:
> It is doable and probably better on the long run, but so far I have been reluctant on touching this side until this part is stable.

Should we add a TODO comment for this?

Reply:
> > It is doable and probably better on the long run, but so far I have been reluctant on touching this side until this part is stable.
> 
> Should we add a TODO comment for this?

I think so.

Reply:
I refactored a bit this part and added a TODO to add function generation capability.

Reply:
Thanks.